xbmc/utils/URIUtils.cpp

```diff
@@ -455,6 +455,48 @@ std::string URIUtils::GetBasePath(const std::string& strPath)
   return strDirectory;
 }
 
+std::string URIUtils::GetDiscBase(const std::string& file)
+{
+  std::string discFile;
+  if (IsBlurayPath(file))
+    discFile = GetBlurayFile(file);
+  else
+    discFile = file;
+
+  std::string parent{GetParentPath(discFile)};
+  std::string parentFolder{parent};
+  RemoveSlashAtEnd(parentFolder);
+  parentFolder = GetFileName(parentFolder);
+  if (StringUtils::EqualsNoCase(parentFolder, "VIDEO_TS") ||
+      StringUtils::EqualsNoCase(parentFolder, "BDMV"))
+    return GetParentPath(parent); // go back up another one
+  return parent;
+}
+
+std::string URIUtils::GetDiscBasePath(const std::string& file)
+{
+  std::string base{GetDiscBase(file)};
+  if (IsDiscImage(base))
+    return GetDirectory(base);
+  return base;
+}
+
+std::string URIUtils::GetBlurayFile(const std::string& path)
+{
+  if (IsBlurayPath(path))
+  {
+    const CURL url(path);
+    const CURL url2(url.GetHostName()); // strip bluray://
+    if (url2.IsProtocol("udf"))
+      // ISO
+      return url2.GetHostName(); // strip udf://
+    if (url.IsProtocol("bluray"))
+      // BDMV
+      return url2.Get() + "BDMV/index.bdmv";
```

Comment:
I know this was existing code, but please use `URIUtils::AddFileToFolder()` to join paths, e.g. `URIUtils::AddFileToFolder(url2.Get(), "BDMV", "index.bdmv")`